### PR TITLE
fix: [PL-25758]: Fixed check on Managed role while querying for Account Viewer roles for Migration

### PR DESCRIPTION
**Pull request by gupta-utkarsh**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34107)

### CODE REVIEW

File: access-control/service/src/main/java/io/harness/accesscontrol/roleassignments/migration/AccountBasicRoleAssignmentAdditionMigration.java [hunk 1]
```diff
@@ -62,27 +64,34 @@ public void migrate() {
...
-        String accountId =
-            scopeService.buildScopeFromScopeIdentifier(roleAssignment.getScopeIdentifier()).getInstanceId();
-        try {
-          featureFlagForAccounts.computeIfAbsent(
-              accountId, accId -> featureFlagHelperService.isEnabled(FeatureName.ACCOUNT_BASIC_ROLE, accId));
-          if (Boolean.TRUE.equals(featureFlagForAccounts.get(accountId))) {
-            RoleAssignmentDBO newRoleAssignmentDBO = buildRoleAssignmentDBO(roleAssignment);
-            roleAssignmentRepository.save(newRoleAssignmentDBO);
-            roleAssignmentRepository.updateById(roleAssignment.getId(), update(RoleAssignmentDBOKeys.managed, false));
+        if (!roleAssignment.isManaged()) {
```
Comment (Jimit09Gandhi) [inline access-control/service/src/main/java/io/harness/accesscontrol/roleassignments/migration/AccountBasicRoleAssignmentAdditionMigration.java:77]: The check should be just roleAssignment.isManaged() as we want to migrate Account Viewer role assignments which are managed.

Comment (Jimit09Gandhi) [inline access-control/service/src/main/java/io/harness/accesscontrol/roleassignments/migration/AccountBasicRoleAssignmentAdditionMigration.java:77]: The check should be just roleAssignment.isManaged() as we want to migrate Account Viewer role assignments which are managed.

**Comment by gupta-utkarsh** [comment]
trigger messagecheck

**Review by reetika04 — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34107)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34107#-:-N46OvRS9KZ2TvxdzQ-k:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**